ts/src/pro/xt.ts

```diff
@@ -0,0 +1,1079 @@
+        const method = this.safeString (params, 'method', defaultMethod);
+        const name = method + '@' + market['id'];
+        return await this.subscribe (name, 'public', 'watchTicker', market, params);
+    }
+
+    async watchTickers (symbols: string[] = undefined, params = {}) {
+        /**
+         * @method
+         * @name xt#watchTicker
+         * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @see https://doc.xt.com/#websocket_publicallTicker
+         * @see https://doc.xt.com/#futures_market_websocket_v2allTicker
+         * @see https://doc.xt.com/#futures_market_websocket_v2allAggTicker
+         * @param {string} symbol not used by xt watchTickers
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @param {string} params.method 'agg_tickers' (contract only) or 'tickers', default = 'tickers' - the endpoint that will be streamed
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        const options = this.safeValue (this.options, 'watchTickers');
+        const defaultMethod = this.safeString (options, 'method', 'tickers');
+        const name = this.safeString (params, 'method', defaultMethod);
+        return await this.subscribe (name, 'public', 'watchTickers', undefined, params);
+    }
+
+    async watchOHLCV (symbol: string, timeframe = '1m', since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name hitbtc#watchOHLCV
+         * @description watches historical candlestick data containing the open, high, low, and close price, and the volume of a market
+         * @see https://doc.xt.com/#websocket_publicsymbolKline
+         * @see https://doc.xt.com/#futures_market_websocket_v2symbolKline
+         * @param {string} symbol unified symbol of the market to fetch OHLCV data for
+         * @param {string} timeframe 1m, 3m, 5m, 15m, 30m, 1h, 2h, 4h, 6h, 8h, 12h, 1d, 3d, 1w, or 1M
+         * @param {int|undefined} since not used by xt watchOHLCV
+         * @param {int|undefined} limit not used by xt watchOHLCV
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @returns {[[int]]} A list of candles ordered as timestamp, open, high, low, close, volume
+         */
+        const market = this.market (symbol);
+        const name = 'kline@' + market['id'] + ',' + timeframe;
+        return await this.subscribe (name, 'public', 'watchOHLCV', market, params);
+    }
+
+    async watchTrades (symbol: string = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name xt#watchTrades
+         * @description get the list of most recent trades for a particular symbol
```

Comment:
yeah definetely the python version isn't working:
```
p xt watchTrades "BTC/USDT" --verbose
Python v3.10.9
CCXT v3.1.6
xt.watchTrades(BTC/USDT)
2023-05-30T14:13:19.696Z connecting to wss://stream.xt.com/public with timeout 10000 ms
2023-05-30T14:13:20.547Z connected
2023-05-30T14:13:20.547Z ping loop
2023-05-30T14:13:20.547Z sending ping
2023-05-30T14:13:20.547Z sending {'method': 'subscribe', 'id': '1685455999695trade@btc_usdt', 'params': ['trade@btc_usdt']}
2023-05-30T14:13:20.755Z message pong
2023-05-30T14:13:20.755Z message {"id":"1685455999695trade@btc_usdt","code":0,"msg":"SUCCESS","method":"subscribe"}
2023-05-30T14:13:20.950Z message {"topic":"trade","event":"trade@btc_usdt","data":{"s":"btc_usdt","i":"233849228424681857","t":1685456000765,"p":"27809.80","q":"0.000945","b":false}}
2023-05-30T14:13:21.350Z message {"topic":"trade","event":"trade@btc_usdt","data":{"s":"btc_usdt","i":"233849229922048410","t":1685456001122,"p":"27809.80","q":"0.197065","b":false}}
2023-05-30T14:13:21.574Z message {"topic":"trade","event":"trade@btc_usdt","data":{"s":"btc_usdt","i":"233849230920292737","t":1685456001360,"p":"27809.80","q":"0.000140","b":false}}
```

Reply:
It looks like it's working for me

```
% py xt watchTrades "BTC/USDT" --verbose
Python v3.11.3
CCXT v3.1.6
xt.watchTrades(BTC/USDT)
2023-06-01T21:37:52.678Z connecting to wss://stream.xt.com/public with timeout 10000 ms
2023-06-01T21:37:53.950Z connected
2023-06-01T21:37:53.951Z ping loop
2023-06-01T21:37:53.951Z sending ping
2023-06-01T21:37:53.951Z sending {'method': 'subscribe', 'id': '1685655472678trade@btc_usdt', 'params': ['trade@btc_usdt']}
2023-06-01T21:37:54.314Z message pong
2023-06-01T21:37:54.314Z message {"id":"1685655472678trade@btc_usdt","code":0,"msg":"SUCCESS","method":"subscribe"}
2023-06-01T21:37:54.419Z message {"topic":"trade","event":"trade@btc_usdt","data":{"s":"btc_usdt","i":"234685880726493569","t":1685655474218,"p":"26922.86","q":"0.000068","b":false}}
[{'amount': 6.8e-05,
  'cost': 1.83075448,
  'datetime': '2023-06-01T21:37:54.218Z',
  'fee': {'cost': None, 'currency': None, 'rate': None},
  'fees': [{'cost': None, 'currency': None, 'rate': None}],
  'id': '234685880726493569',
  'info': {'b': False,
           'i': '234685880726493569',
           'p': '26922.86',
           'q': '0.000068',
           's': 'btc_usdt',
           't': 1685655474218},
  'order': None,
  'price': 26922.86,
  'side': 'buy',
  'symbol': 'BTC/USDT',
  'takerOrMaker': 'taker',
  'timestamp': 1685655474218,
  'type': None}]
  ```